### PR TITLE
build: consolidate @angular-devkit/build-angular to the root package.json

### PR DESCRIPTION
**Pull request by IgorMinar**
This allows us to update the version of the package in a single place for all tests.

Notable exemption of this is aio which currently doesn't depend on anything installed in the root.

todos:
- [ ] update integration/README.md
- [x] rebase on top of #33968

### CODE REVIEW

File: integration/bazel-schematics/test.sh [hunk 1]
```diff
@@ -52,7 +52,7 @@ function testNonBazel() {
   # disable CLI's version check (if version is 0.0.0, then no version check happens)
   yarn --cwd node_modules/@angular/cli version --new-version 0.0.0 --no-git-tag-version
   # re-add build-angular
-  yarn add --dev @angular-devkit/build-angular@0.900.0-rc.3
+  yarn add --dev file:../../node_modules/@angular-devkit/build-angular
```
Comment (gkalpak) [inline integration/bazel-schematics/test.sh]: You need another `../` in the path.
Also, do we need [this workaround][1] for local deps caching here too? Or is it only a problem with `yarn install` and not `yarn add`?

[1]: https://github.com/angular/angular/blob/c4ccb09cc/integration/run_tests.sh#L39-L40

File: integration/cli-hello-world-ivy-compat/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
     "zone.js": "file:../../dist/zone.js-dist/zone.js"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "^0.900.0-rc.3",
+    "@angular-devkit/build-angular": "file:../../node_modules/@angular-devkit/build-angular",
```
Comment (gkalpak) [inline integration/cli-hello-world-ivy-compat/package.json]: This should not work per [this discussion][1] 😕
How come it does?

[1]: https://github.com/angular/angular/pull/32957#discussion_r336056677

**Review by gkalpak — CHANGES_REQUESTED** [review]
CI is a bit unhappy, but otherwise looks great :smiley:

File: integration/bazel-schematics/test.sh (hunk 1 above)
Comment (IgorMinar) [reply integration/bazel-schematics/test.sh]: I was missing ../

File: integration/cli-hello-world-ivy-compat/package.json (hunk 2 above)
Comment (filipesilva) [reply integration/cli-hello-world-ivy-compat/package.json]: I think @clydin added some better monorepo support recently, so maybe that was it.

Comment (clydin) [reply integration/cli-hello-world-ivy-compat/package.json]: I'm not overly familiar with the setup here but I did fix a variety of package resolution edge cases recently in both the `@angular-devkit/build-angular` and `@ngtools/webpack` packages.

Comment (IgorMinar) [reply integration/cli-hello-world-ivy-compat/package.json]: as far as I know the only reason why this didn't work in angular/angular in the past was an issue in bazel which has been fixed already.

**Review by gkalpak — APPROVED** [review]
Other than https://github.com/angular/angular/pull/34002#discussion_r349873213 and the conflicts, lgtm 👌

**Comment by IgorMinar** [comment]
rebased

**Comment by IgorMinar** [comment]
the ci is still failing. looks like some kind of typings issue :-/

File: package.json [hunk 3]
```diff
@@ -134,6 +134,7 @@
   "// 3": "when updating @bazel/bazel version you also need to update the RBE settings in .bazelrc (see https://github.com/angular/angular/pull/27935)",
   "devDependencies": {
     "@angular/cli": "^9.0.0-rc.3",
+    "@angular-devkit/build-angular": "^0.900.0-rc.3",
```
Comment (gkalpak) [inline package.json]: Nit: Package managers put this above `@angular/cli` :grin:

File: integration/bazel-schematics/test.sh [hunk 4]
```diff
@@ -52,7 +52,7 @@ function testNonBazel() {
   # disable CLI's version check (if version is 0.0.0, then no version check happens)
   yarn --cwd node_modules/@angular/cli version --new-version 0.0.0 --no-git-tag-version
   # re-add build-angular
-  yarn add --dev @angular-devkit/build-angular@0.900.0-rc.3
+  yarn add --dev file:../../../node_modules/@angular-devkit/build-angular
```
Comment (gkalpak) [inline integration/bazel-schematics/test.sh]: Do we need [this workaround][1] for local deps caching here too? Or is it only a problem with `yarn install` and not `yarn add`?

[1]: https://github.com/angular/angular/blob/c4ccb09cc/integration/run_tests.sh#L39-L40

Comment (IgorMinar) [reply integration/bazel-schematics/test.sh]: afaik only `yarn install` has that problem, I tested it and `add` works correctly.

**Comment by IgorMinar** [comment]
merge-assistance: global approval

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
@IgorMinar are you allowed to global approve your own PRs ?? Or is @gkalpak's approval enough to enable your global approval?

**Review by josephperrott — APPROVED** [review]
LGTM

**Pull request closed without merging**